homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,229 @@
+        """Import config from yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_API_KEY] == import_config[CONF_API_KEY]:
+                return self.async_abort(reason="already_configured")
+        return self.async_create_entry(
+            title="LastFM",
+            data={
+                CONF_API_KEY: import_config[CONF_API_KEY],
+                CONF_MAIN_USER: None,
+                CONF_USERS: import_config[CONF_USERS],
+            },
+        )
+
+
+class LastFmOptionsFlowHandler(OptionsFlow, LastFmFlowHandler):
+    """LastFm Options flow handler."""
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Initialize LastFM Options flow."""
+        super().__init__(dict(entry.data))
+        self.entry = entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize form."""
+        valid_users = self.data[CONF_USERS]
+        errors = {}
+        if user_input is not None:
+            valid_users = []
+            for user in user_input[CONF_USERS]:
+                lastfm_user = self._get_lastfm_user(user)
+                lastfm_errors = self._validate_lastfm_user(lastfm_user)
+                if lastfm_errors:
+                    errors = lastfm_errors
+                else:
+                    valid_users.append(user)
+            if not errors:
+                await self.hass.config_entries.async_unload(self.entry.entry_id)
+                for username in self.data[CONF_USERS]:
+                    if username not in user_input[CONF_USERS] and (
+                        entity_id := er.async_get(self.hass).async_get_entity_id(
+                            Platform.SENSOR, DOMAIN, f"sensor.lastfm_{username}"
+                        )
+                    ):
+                        er.async_get(self.hass).async_remove(entity_id)
+                await self.hass.config_entries.async_reload(self.entry.entry_id)
+                return self.async_create_entry(
+                    title="LastFM",
+                    data={
+                        **self.data,
+                        CONF_USERS: user_input[CONF_USERS],
+                    },
+                )
+        if self.data[CONF_MAIN_USER]:
+            try:
+                main_user = self._get_lastfm_user(self.data[CONF_MAIN_USER])
+                friends: Sequence[SelectOptionDict] = [
+                    {"value": str(friend.name), "label": str(friend.get_name(True))}
+                    for friend in main_user.get_friends()
+                ]
+            except WSError:
+                friends = []
+        else:
+            friends = []
+        return self.async_show_form(
+            step_id="init",
+            errors=errors,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERS, default=valid_users): SelectSelector(
```

Comment:
Should not be a default but instead use `self.add_suggested_values_to_schema` with the schema, [example](https://github.com/gjohansson-ST/core/blob/dev/homeassistant/components/workday/config_flow.py#L288)